src/essreflectometry/amor/calibrations.py

```diff
@@ -19,39 +21,33 @@ def supermirror_calibration(
     data_array:
         Data array to get q-bins/values from.
     m_value:
-        m-value for the supermirror. Defaults to 5.
+        m-value for the supermirror.
     critical_edge:
-        Supermirror critical edge. Defaults to 0.022 1/angstrom.
+        Supermirror critical edge.
     alpha:
-        Supermirror alpha value. Defaults to 0.25 / 0.088 angstrom.
-
+        Supermirror alpha value.
     Returns
     -------
     :
         Calibrated supermirror measurement.
     """
-    if m_value is None:
-        m_value = sc.scalar(5, unit=sc.units.dimensionless)
-    if critical_edge is None:
-        critical_edge = 0.022 * sc.Unit('1/angstrom')
-    if alpha is None:
-        alpha = sc.scalar(0.25 / 0.088, unit=sc.units.angstrom)
     calibration = calibration_factor(data_array, m_value, critical_edge, alpha)
```

Comment:
Consider this:
- Remove `supermirror_calibration`
- make `calibration_factor` a provider
- remove `normalize_sample_by_counts` and `normalize_reference_by_counts`
- use `normalize_by_counts` for the sample
- add `normalize_reference`, which normalizes by counts and has the calibration factor as another input, i.e., normalizes by both counts and calibration. This might actually be cheaper if you combine the two factors, since there will be only a single "large" operation?

Reply:
I have a question about this.
The supermirror calibration is in the amor module, but `normalize_reference` is in the generic reflectometry code.
I don't think it's good to have the generic reflectometry code depend on amor.
Should we move the supermirror calibration to reflectometry or the normalization step into amor?
We've already moved the supermirror parameter providers into reflectometry.types so maybe it makes most sense to move the entire supermirror logic into reflectometry also.
What do you think?

Reply:
> I don't think it's good to have the generic reflectometry code depend on amor.

Yes.

> Should we move the supermirror calibration to reflectometry or the normalization step into amor?

Depends, is it generic?

> We've already moved the supermirror parameter providers into reflectometry.types so maybe it makes most sense to move the entire supermirror logic into reflectometry also.
> What do you think?

Not necessarily: By having the parameters in a generic place, the workflow can say "I need supermirror calibration". One could then have different providers that use this, in different, e.g., instrument-specific submodules. This does not imply that the generic reflectometry code depends on amor. Only if you make an amor pipeline then you will use the amor providers. On other instruments you would need to use a different provider.

If it is probably generic, then move it.